src/Client.php

```diff
@@ -394,15 +394,29 @@ public function send($method, $payload, $path, $params = [], array $headers = []
 
         $uri = urldecode($location->getPath());
 
-        // Rewind stream
-        // see: https://github.com/guzzle/psr7/issues/38
-        $response->getBody()->rewind();
-
         // Unserialize response body
-        $content = json_decode($response->getBody()->getContents(), true);
+        $content = json_decode((string) $response->getBody(), true) ?: [];
 
         // Build expected resource
-        $resource = $this->factory->create($uri, $content);
+        $resource = $this->factory->create($uri, []);
```

Comment:
The above json_decode() value is used only inside `if`, not inside `else` or I'm missing something.

else is from L414-416... the above value is not used... there you create a new value for the variable $content.